code/modules/mob/mob_helpers.dm

```diff
@@ -286,10 +286,10 @@
 	if(hud_used && hud_used.action_intent)
 		hud_used.action_intent.icon_state = "[a_intent]"
 
-///Checks if the mob is able to see or not. eye_blind is temporary blindness, the trait is if they're permanently blind.
+///Checks if the mob is able to see or not. eye_blind is temporary blindness which can be countered by true sight, the trait is if they're permanently blind.
 /mob/proc/is_blind()
 	SHOULD_BE_PURE(TRUE)
-	return eye_blind ? TRUE : HAS_TRAIT(src, TRAIT_BLIND)
+	return eye_blind ? HAS_TRAIT(src, TRAIT_TRUE_SIGHT) : HAS_TRAIT(src, TRAIT_BLIND)
```

Comment:
I really dislike these chains of ifs and elses to determine simple things such as if someone can hear/see or not.
This makes trying to react to the event of gaining or losing sight impossible without a refactor, which gets harder to disentangle the more complexity these checks get.

We need to substitute `is_blind()` with a single trait, keep track of the sources and add and remove to it properly instead of stacking sources and exceptions with no way to report the event.